src/connection/query.ts

```diff
@@ -12,28 +12,57 @@ export class Query<T> {
   private correlationId: string;
   private sql: string;
   private isPrepared: boolean = false;
-  private parameters: any[];
+  private parameters: any[]|undefined;
   private rowsToFetch: number = 100;
   private isCLCommand: boolean;
   private state: QueryState = QueryState.NOT_YET_RUN;
 
-  constructor(private job: SQLJob, query: string, opts: QueryOptions = {isClCommand: false, parameters: undefined}) {
+  public autoClose: boolean | undefined;
+
+  constructor(private job: SQLJob, query: string, opts: QueryOptions = { isClCommand: false, parameters: undefined, autoClose: false }) {
     this.job = job;
     this.isPrepared = (undefined !== opts.parameters);
     this.parameters = opts.parameters;
     this.sql = query;
     this.isCLCommand = opts.isClCommand;
+    this.autoClose = opts.autoClose;

```

Comment:
Personal preference and precedence in this part is that boolean member vars start with "is"... So `this.isAutoClose`

Reply:
I agree that this should be more explicit, but I don't know if `is` is right. I will try `shouldAutoClose`. Lmk what you think

Reply:
`shouldAutoClose` is perfect